routes/index.js

```diff
@@ -91,9 +87,28 @@ module.exports = function(utils, env, nunjucksEnv, appName) {
         userbar: userbarEndpoint,
         webmaker: webmaker,
         makedetails: makedetails,
-        editorURL: editorURL + '/index.html' + queryString,
-        editorHOST: editorHOST
-      });
+        editorHOST: editorHOST,
+        OAUTH_CLIENT_ID: oauthClientId,
+        OAUTH_AUTHORIZATION_URL: oauthUrl
+      };
+
+      // We add the localization code to the query params through a URL object
+      // and set search prop to nothing forcing query to be used during url.format()
+      var urlObj = url.parse(req.url, true);
+      urlObj.search = "";
+      urlObj.query["locale"] = req.localeInfo.lang;
+      var thimbleUrl = url.format(urlObj);
+
+      // We forward query string params down to the editor iframe so that
+      // it's easy to do things like enableExtensions/disableExtensions
+      options.editorURL = editorURL + '/index.html' + (url.parse(thimbleUrl).search || '');
+
+      if (req.user) {
+        options.username = req.user.username;
+        options.gravatar = req.user.avatar;
```

Comment:
`req.user.avatar` is correct. `avatar` is what the ID server sends back, and `gravatar` is used in a few places. Making it consistent is probably a good idea.